pkg/cluster/private_registry.go

```diff
@@ -119,25 +124,28 @@ func GeneratePrivateRegistryEncodedDockerConfig(cluster *v3.Cluster, secretListe
 	// For RKE2 with a cluster level registry configured, this is the
 	// only reference to the registry URL available on the v3.Cluster.
 	// Without it, we cannot generate the registry credentials (.dockerconfigjson)
-	registryURL := cluster.GetSecret(v3.ClusterPrivateRegistryURL)
-	if registryURL == "" {
-		return "", "", nil
+	v2ProvRegistryURL := cluster.GetSecret(v3.ClusterPrivateRegistryURL)
+	// If we don't have a secretName nor a downstream PrivateRegistryURL on the v2Prov we return the RKE1/default registry URL.
+	if v2ProvRegistryURL == "" {
+		return rkeClusterURLOrGlobalSystemDefault, "", nil
 	}
 
-	if registrySecretName == "" { // no private registry configured
-		return registryURL, "", nil
+	// if we have a v2Prov registry URL, and we don't have a secret we just return the downstream URL.
```

Comment:
Just to add a bit more explanation, this condition will happen if the v2 prov registry exists, but is _unauthenticated_.